modules/text2vec-cohere/vectorizer/class_settings.go

```diff
@@ -0,0 +1,192 @@
+)
+
+var availableCohereModels = []string{"small", "medium", "large"}
+var availableTruncates = []string{"NONE", "LEFT", "RIGHT"}
+
+type classSettings struct {
+	cfg moduletools.ClassConfig
+}
+
+func NewClassSettings(cfg moduletools.ClassConfig) *classSettings {
+	return &classSettings{cfg: cfg}
+}
+
+func (ic *classSettings) PropertyIndexed(propName string) bool {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return DefaultPropertyIndexed
+	}
+
+	vcn, ok := ic.cfg.Property(propName)["skip"]
+	if !ok {
+		return DefaultPropertyIndexed
+	}
+
+	asBool, ok := vcn.(bool)
+	if !ok {
+		return DefaultPropertyIndexed
+	}
+
+	return !asBool
+}
+
+func (ic *classSettings) VectorizePropertyName(propName string) bool {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return DefaultVectorizePropertyName
+	}
+	vcn, ok := ic.cfg.Property(propName)["vectorizePropertyName"]
+	if !ok {
+		return DefaultVectorizePropertyName
+	}
+
+	asBool, ok := vcn.(bool)
+	if !ok {
+		return DefaultVectorizePropertyName
+	}
+
+	return asBool
+}
+
+func (ic *classSettings) Model() string {
+	return ic.getProperty("model", DefaultCohereModel)
+}
+
+func (ic *classSettings) Truncate() string {
+	return ic.getProperty("truncate", DefaultTruncate)
+}
+
+func (ic *classSettings) VectorizeClassName() bool {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return DefaultVectorizeClassName
+	}
+
+	vcn, ok := ic.cfg.Class()["vectorizeClassName"]
+	if !ok {
+		return DefaultVectorizeClassName
+	}
+
+	asBool, ok := vcn.(bool)
+	if !ok {
+		return DefaultVectorizeClassName
+	}
+
+	return asBool
+}
+
+func (ic *classSettings) Validate(class *models.Class) error {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return errors.New("empty config")
+	}
+
+	model := ic.Model()
+	if !ic.validateCohereSetting(model, availableCohereModels) {
+		return errors.Errorf("wrong Cohere model name, available model names are: %v", availableCohereModels)
+	}
+	truncate := ic.Truncate()
+	if !ic.validateCohereSetting(truncate, availableTruncates) {
+		return errors.Errorf("wrong truncate type, available types are: %v", availableTruncates)
+	}
+
+	err := ic.validateIndexState(class, ic)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ic *classSettings) validateCohereSetting(value string, availableValues []string) bool {
+	for i := range availableValues {
+		if value == availableValues[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func (ic *classSettings) getProperty(name, defaultValue string) string {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return defaultValue
+	}
+
+	model, ok := ic.cfg.Class()[name]
+	if ok {
+		asString, ok := model.(string)
+		if ok {
+			return strings.ToLower(asString)
```

Comment:
you should remove `strings.ToLower` method here and just return a string, it breaks setting of `truncate` value as the value being returned is always lowercased and you are checking against `availableTruncates` array which holds upper cased string values